src/WebJobs.Script.WebHost/Middleware/ScriptHostRequestServiceProviderMiddleware.cs

```diff
@@ -12,19 +12,19 @@ namespace Microsoft.Azure.WebJobs.Script.WebHost.Middleware
     public class ScriptHostRequestServiceProviderMiddleware
     {
         private readonly RequestDelegate _next;
+        private readonly WebJobsScriptHostService _scriptHostService;
 
-        public ScriptHostRequestServiceProviderMiddleware(RequestDelegate next)
+        public ScriptHostRequestServiceProviderMiddleware(RequestDelegate next, WebJobsScriptHostService scriptHostService)
         {
             _next = next;
+            _scriptHostService = scriptHostService;
```

Comment:
Curious -- is there a perf improvement by moving this to the constructor?

Reply:
Well, doing the DI resolution on Invoke incurs a DI resolution per invocation